src/webui/www/private/shareratio.html

```diff
@@ -0,0 +1,168 @@
+            };
+
+            // select current values
+            if (hashesList.length === 1) {
+                var hash = hashesList[0];
+                // using double equals to compare string and int
+                if ((values.ratioLimit === UseGlobalLimit) && (values.seedingTimeLimit === UseGlobalLimit)) {
+                    // use default option
+                    setSelectedRadioValue('shareLimit', 'default');
+                }
+                else if ((values.maxRatio == NoLimit) && (values.maxSeedingTime == NoLimit)) {
+                    setSelectedRadioValue('shareLimit', 'none');
+                }
+                else {
+                    setSelectedRadioValue('shareLimit', 'custom');
+                    if (values.ratioLimit >= 0) {
+                        $('setRatio').set('checked', true);
+                        $('ratio').set('value', values.ratioLimit);
+                    }
+                    if (values.seedingTimeLimit >= 0) {
+                        $('setMinutes').set('checked', true);
+                        $('minutes').set('value', values.seedingTimeLimit);
+                    }
+                }
+            }
+
+            shareLimitChanged();
+
+            $('default').focus();
+            $('save').addEvent('click', function(e) {
+                new Event(e).stop();
+
+                if (!isFormValid()) {
+                    return false;
+                }
+
+                var shareLimit = getSelectedRadioValue('shareLimit');
+                var ratioLimitValue = 0.00;
+                var seedingTimeLimitValue = 0;
+
+                if (shareLimit === 'default') {
+                    ratioLimitValue = seedingTimeLimitValue = UseGlobalLimit;
+                }
+                else if (shareLimit === 'none') {
+                    ratioLimitValue = seedingTimeLimitValue = NoLimit;
+                }
+                else if (shareLimit === 'custom') {
+                    if ($('setRatio').get('checked'))
+                        ratioLimitValue = $('ratio').get('value');
+                    if ($('setMinutes').get('checked'))
+                        seedingTimeLimitValue = $('minutes').get('value');
+                }
+                else return false;
+
+                new Request({
+                    url: 'api/v2/torrents/setShareLimit',
+                    method: 'post',
+                    data: {
+                        hashes: hashesList.join('|'),
+                        ratioLimit: ratioLimitValue,
+                        seedingTimeLimit: seedingTimeLimitValue
+                    },
+                    onComplete: function () {
+                        window.parent.closeWindows();
+                    }
+                }).send();
+            });
+        });
+
+        function getSelectedRadioValue(name) {
+            var radios = document.getElementsByName(name);
+
+            for (var i = 0; i < radios.length; i++) {
```

Comment:
`++i` and instances below

Reply:
In this context the results are identical, and `i++` is a lot more common in JavaScript. We have 39 instances of `i++` in our js codebase, but only 1 instance of `++i`.

Reply:
I'm merely pointing out the coding style, until the coding guidelines extends to JS, I can't/won't force you to change it.

>In this context the results are identical, and i++ is a lot more common in JavaScript. 

Of course and I often wonder why that is the case in JS world given `++i`, `i++`has the same semantics as in C++...